api/grpcserver/transaction_service.go

```diff
@@ -236,7 +234,7 @@ func (s TransactionService) TransactionsStateStream(in *pb.TransactionsStateStre
 			// In order to read transactions, we first need to read layer blocks
 			layerObj, err := s.mesh.GetLayer(layer.LayerID)
 			if err != nil {
-				s.logger.With().Error("error reading layer data for updated layer", layer.LayerID, log.Err(err))
+				ctxzap.Error(stream.Context(), "error reading layer data for updated layer", layer.LayerID.Field().Zap(), zap.Error(err))
```

Comment:
Instead of extending `log.Field` with a `Zap()` method I think it would be better to have `LayerID` implement the `zapcore.ObjectMarshaler` interface.

This way we don't need to extend the (hopefully soon to be removed) `log` package and the field is evaluated lazily instead of eagerly (in case the log isn't even printed).

Reply:
Yes, 100% agree. I added this `Zap()` method as a shortcut before we migrate to not have to re-implement (and avoid having 2 implementations for log field serialization).